File: JULY-2025/13-07-2025/p2410.dart
class Solution {
  int matchPlayersAndTrainers(List<int> players, List<int> trainers) {
  players.sort();
  trainers.sort();
  int count = 0, i = 0, j = 0;
  while (i < players.length && j < trainers.length) {
    if (players[i] <= trainers[j]) {
      count++;
      i++;
    }
    j++;
  }
  return count;
}
}

File: JULY-2025/13-07-2025/p2410.cs
public class Solution {
    public int MatchPlayersAndTrainers(int[] players, int[] trainers) {
        // Sort both arrays to apply greedy matching
        Array.Sort(players);
        Array.Sort(trainers);

        int i = 0; // pointer for players
        int j = 0; // pointer for trainers
        int ans = 0; // count of successful matches

        // Use two pointers to find optimal matches
        while (i < players.Length && j < trainers.Length) {
            if (players[i] <= trainers[j]) {
                // Match found: current player can be trained by current trainer
                ans++;
                i++;
                j++;
            } else {
                // Current trainer too weak, try next stronger trainer
                j++;
            }
        }

        // Return the total number of matched player-trainer pairs
        return ans;
    }
}